RecoTracker/LSTCore/src/alpaka/NeuralNetwork.h

```diff
@@ -58,6 +58,12 @@ namespace lst::t5dnn {
     float eta4 = alpaka::math::abs(acc, mdsInGPU.anchorEta[mdIndex4]);  // outer T3 anchor hit 4 eta (t3_2_eta)
     float eta5 = alpaka::math::abs(acc, mdsInGPU.anchorEta[mdIndex5]);  // outer T3 anchor hit 5 eta (t3_4_eta)
 
+    float phi1 = alpaka::math::abs(acc, mdsInGPU.anchorPhi[mdIndex1]);  // inner T3 anchor hit 1 phi (t3_0_phi)
+    float phi2 = alpaka::math::abs(acc, mdsInGPU.anchorPhi[mdIndex2]);  // inner T3 anchor hit 2 phi (t3_2_phi)
+    float phi3 = alpaka::math::abs(acc, mdsInGPU.anchorPhi[mdIndex3]);  // inner T3 anchor hit 3 phi (t3_4_phi)
+    float phi4 = alpaka::math::abs(acc, mdsInGPU.anchorPhi[mdIndex4]);  // outer T3 anchor hit 4 phi (t3_2_phi)
+    float phi5 = alpaka::math::abs(acc, mdsInGPU.anchorPhi[mdIndex5]);  // outer T3 anchor hit 5 phi (t3_4_phi)
```

Comment:
it seems that it would be useful to get signed deltaPhi before computing the abs. To at least avoid a wrap-around mess

Reply:
Oh yeah, good point. Thank you

Reply:
Looking at the CI plots, I didn't find any striking pathologies in plots vs phi for efficiency.
although there are two bins near 0 and π in T5 fr vs phi that may be an indication of 
![image](https://github.com/user-attachments/assets/89cf49d4-f485-4e54-a54a-97bfbfb5721b)

Reply:
I thought about this more, but are there any instances you see where there would be a wrap around problem? Since I take the absolute value of all phi's first, I feel like this should be fine? If you have one hit near pi and the next hit near -pi for example, I take the absolute value of both first so you would get the correct small separation.

Reply:
The problem is the continuity is easily violated around 0 and π.
it could be `abs` is a problem around 0, e.g. `0,0.1,-0.15,0.2` is not a natural pattern compared to its abs value.
Or around π e.g. `π-0.1,π,π-0.1` is not good but `π-0.1,π,-π+0.1` is OK.

Reply:
I'm finding that using the signed differences results in a bit worse performance. Given that this PR is good-to-go, and this issue also involves the other variables (signed difference of eta for ex.) can we make an issue for this and address it in another PR?

Reply:
just to be sure, was the test to use e.g. `deltaPhi(anchorPhi[mdIndex2], anchorPhi[mdIndex1])` instead of what it is now `abs(anchorPhi[mdIndex2])-abs(anchorPhi[mdIndex1])`, or something else?

Numerically this should affect only the small regions near 0 and π. Is the performance getting worse there or in some other regions?

Reply:
> deltaPhi(anchorPhi[mdIndex2], anchorPhi[mdIndex1])

in terms of our methods names `phi_mpi_pi(anchorPhi[mdIndex2]-  anchorPhi[mdIndex1])`

Reply:
I implemented my own delta phi just to be sure that I used the same function during training (see below), but I don't think the distributions are going to be exactly the same, right? If both phi values are positive then yes of course, but if both are negative and you have for example -20 degrees to -30 degrees then you would get -10 degrees for deltaPhi(phi2, phi1) vs. positive 10 degrees for abs(-30) - abs(-20).

![Screenshot 2024-11-07 at 7 55 02 PM](https://github.com/user-attachments/assets/af72a5f0-4c22-4cbb-a9d7-efd3bec11f24)

Reply:
@slava77 I figured the issue out I think. Right now, I'm normalizing the phi input variables by dividing by pi, but when I moved to the signed deltas I stopped normalizing them because their distributions were already small. Removing those normalizations caused the DNN to rely on the phi variables more which reduced the displaced track efficiency slightly at the improvement of a lower FR. If I boosted the displaced track weight up more in-turn maybe it would have sorted this out, but I'm just going to leave the normalizations there. I'll add the update commit tomorrow.

Reply:
Fixed, had to increase the weight to 6x either way.